### PR TITLE
GitHub authentication rework and MongoDB Integration

### PR DESCRIPTION
**Pull request by profet23**
Reworks how end user authentication works as well as adding MongoDB Integration.

### CODE REVIEW

File: MSBLOC.Web/Controllers/AccountController.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                var repo = repositories.FirstOrDefault(r => r.Id == savedRepo.Id);
+                if (repo == null) continue;
+                repo.Id = savedRepo.Id;
+                repo.Secret = savedRepo.Secret;
+            }
+
+            return View(repositories);
+        }
+
+        [HttpGet]
```
Comment (zippy1981) [inline MSBLOC.Web/Controllers/AccountController.cs]: There should be a powershell function for this.

Comment (profet23) [reply MSBLOC.Web/Controllers/AccountController.cs]: Don't think so, the secret needs to be created by a logged in (to github) user. If a build system creates the secret we're not really authing anything.  That said, i think I'm going to change this over to use signed JWTs.

Comment (zippy1981) [reply MSBLOC.Web/Controllers/AccountController.cs]: Ok fair enough. Unless there is a GitHub api powershell module with a simple happy path to throw up a GitHub login the way the azure powershell cmdlets can through up a login popup modal.

File: MSBLOC.Web/Services/JsonWebTokenService.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@ public JsonWebTokenService(IOptions<AuthOptions> optionsAccessor)
             _optionsAccessor = optionsAccessor;
         }
 
+        /// <inheritdoc cref="IJsonWebTokenService.CreateToken"/>
```
Comment (StanleyGoldman) [inline MSBLOC.Web/Services/JsonWebTokenService.cs]: 1. `cref="IJsonWebTokenService.CreateToken"` is redundant here.
2. `inheritdoc` does not actually do anything, unless you provide docs at the source, `IJsonWebTokenService.CreateToken`

File: MSBLOC.Web/Controllers/AccountController.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        {
+            var authProperties = new AuthenticationProperties {RedirectUri = "/"};
+            await HttpContext.SignOutAsync(authProperties);
+            return SignOut(authProperties);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> ListRepositories([FromServices] IGitHubUserClientFactory gitHubUserClientFactory)
+        {
+            //TODO: We need to come up with a caching mechanism for storing the user's repositories as we run this code twice on this request. Once here and once in the `GetTokensForUserRepositoriesAsync()` method.
```
Comment (profet23) [inline MSBLOC.Web/Controllers/AccountController.cs]: This probably needs to be remedied.

Comment (StanleyGoldman) [reply MSBLOC.Web/Controllers/AccountController.cs]: I threw a note into the backlog.

**Pull request merged**